Fix up t/07locale.t

khwilliamson opened the pull request:
These 4 commits change

- [ ]  to use LC_NUMERIC instead of LC_ALL, since the numeric portions of the locale are all that are being tested; 
- [ ] to skip the locale tests if LC_NUMERIC isn't available on the platform.  This will also allow the customization of this file in core blead to be removed
- [ ] to use I18N::Langinfo on recent perl vintages, as localeconv() may be just a stub on some platforms;
- [ ]  and to add some Windows locales to the list of possible locales to try.  This last is because none of the previous list are typically found on Windows boxes, so this test was always getting skipped there. I suppose this could be revised to use t/loc_tools.pl, if available. over the built-in list, which can find a locale to use without resorting to this list.

khwilliamson committed:
version::07locale.t: Use I18N::Langinfo, not POSIX::localeconv()

The former is always present on recent perl versions; the latter might
not be.

khwilliamson committed:
version 07locale.t should use LC_NUMERIC, not ALL

This file is working on LC_NUMERIC only; no need to drag other
categories in.

khwilliamson committed:
07locale.t: Add some Windows locales to try

On various windows machines, none of the locales listed in <DATA> exist.
This commit copies ones from core's loc_tools.pl to give additional
possibilities to try

leonerd commented:
While you're looking at this, you might want to know that core perl keeps a customized version of this file, by adding another `skip` condition at the start:

```diff
--- /home/leo/.cpan/build/version-0.9930-0/t/07locale.t 2023-09-11 18:03:42.000000000 +0100
+++ cpan/version/t/07locale.t   2023-09-21 10:37:36.938698303 +0100
@@ -18,6 +18,8 @@
        skip 'No locale testing for Perl < 5.6.0', 7 if $] < 5.006;
        skip 'No locale testing without d_setlocale', 7
            if(!$Config{d_setlocale});
+       skip 'No locale testing without LC_NUMERIC', 7
+           if($Config{ccflags}) =~ /-DNO_LOCALE_NUMERIC\b/;
 
        # test locale handling
        my $warning = '';
```

If you're fixing things up generally it would be good to get that in sync as well so we can drop a customization.

khwilliamson committed:
07locale.t: Skip locale testing if no LC_NUMERIC

LC_NUMERIC is required for any meaningful locale testing, so if it isn't
available on the platform, or perl is Configured to ignore it, skip
those tests

khwilliamson commented:
I added a 4th commit to this PR that will enable core perl to remove the customization of this file from its blead
